### ISSUE TITLE
Graphical glitches/leftover characters

### ISSUE THREAD

**Issue by rosshettel** [mobile-shell/mosh#798]
Been trying mosh for a couple days now and it's pretty great! Except I've noticed after a little while I get some graphical glitches in the terminal where there are leftover characters that don't clear. This screenshot below is an example of what I'm seeing:
![1__mosh_nasa__mosh-client_](https://cloud.githubusercontent.com/assets/773995/18016068/01c5740e-6b90-11e6-8bfd-40c680f6215c.png)

I'm running the latest version of mosh (1.2.6) on both the server and client. I'm on OS X El Capitan connecting to an Ubuntu 14.04 server, using the latest version of tmux (2.3) and zshell. I am using a theme that uses powerline patched fonts (see: https://github.com/powerline/fonts), I'm not sure if that might be causing the issue.

**Comment by andersk** [comment]
We’ll need some more information to pin this down. Do you have a procedure for reliably reproducing this? See if you can remove as many variables as possible (e.g. can you reproduce without tmux?). If possible, it would help to have a [typescript file](http://linux.die.net/man/1/script) from inside Mosh that reproduces the problem.

**Comment by cgull** [comment]
This has the look of a Unicode character-width issue, but as Anders says, reproducing this reliably and giving us a full description of what and how will help.  Two things I suggest trying:  
* with ssh instead of mosh,
* an older version of tmux (if you can)

**Comment by dequis** [comment]
I have this issue too, likely since the update to mosh 1.2.6. (arch linux server and client. the decision of using arch in the server is not my fault)

ssh doesn't have this issue, and it happens without tmux too.

I get most of the funny text corruption in the twitter window of my irssi, but any fullwidth characters seem to trigger broken behavior. Just copying and pasting "ｆｕｌｌｗｉｄｔｈ　ｔｅｘｔ" several times in a shell seems to desync the cursor position at least.

**Comment by andersk** [comment]
@dequis I wasn’t able to reproduce a problem using your strategy, but if you can, please do provide a typescript file (`mosh SERVER -- script SCRIPTFILE`).

**Comment by dequis** [comment]
Yeah okay this is fiddly. It might not even be mosh, I don't know.

After an hour of testing different things even my main mosh which had this issue all the time stopped doing it. But that mosh is tmux-mosh-tmux, and I can't say for sure that it's mosh anymore. At least i've seen it happen in both xterm and st.

Here's an attempt at a typescript, fwiw: http://dump.dequis.org/9sFeM.txt

**Comment by cgull** [comment]
I can't reproduce this either.  I have been seeing hard-to-pin-down issues on OpenBSD 6.0 that I thought were due to tmux, but now think are elsewhere (probably in wcwidth() on that platform).

@dequis, what terminal emulator are you using?  What OSes are you using at each end of the mosh session?

Another thing that may help:  if you see this, get a screenshot, and in tmux, issue the tmux commands `capture-pane` and `save-buffer screen.txt`.  If you're using tmux at both ends, do that at both ends :)

**Comment by dequis** [comment]
Terminal emulators are xterm and st. OSes at both ends are arch linux.

>Another thing that may help: if you see this, get a screenshot, and in tmux, issue the tmux commands capture-pane and save-buffer screen.txt. If you're using tmux at both ends, do that at both ends :)

Luckily it seems the issue is back and I can do this!

[Screenshot](http://dump.dequis.org/WvtEN.png) (note that this changes every time I do ctrl-L and sometimes when issuing the tmux commands, so it won't be exactly the same as the captures)

Outer tmux: [with -e](http://dump.dequis.org/uyhCw.txt), [without -e](http://dump.dequis.org/-v7fn.txt)

Inner tmux: [with -e](http://dump.dequis.org/2dIEQ.txt), [without -e](http://dump.dequis.org/0Zbn9.txt)

>Also I notice that you're in the ART timezone (Argentina, apparently?), so probably not using US English or one of the other well-tested European locales. Are you using CJK text, too?

I set LANG to `en_US.utf8` everywhere for sanity. I do run into CJK text regularly on twitter (follow everyunicode, fakeunicode, people who use fullwidth text as if it was another font, actual japanese people)
